### ISSUE TITLE
Confusion about the attention implementation.

### ISSUE THREAD

**Issue by xandery-geek** [nttmdlab-nlp/VDocRAG#9]
In the code, the attention implementation for pretraining is eager, rather than flash_attention_2. I am confused as to why flash_attention_2 is not used.

```python
model = VDocRetriever.build(
        model_args,
        training_args,
        cache_dir=model_args.cache_dir,
        trust_remote_code=True,
        torch_dtype=torch_dtype, 
        _attn_implementation='eager' if data_args.pretrain else 'flash_attention_2',
    )
```

**Comment by rtanaka-lab** [comment]
Thank you for the great question. The original FlashAttention implementation does not support the custom attention mask that we used during self-supervised pretraining, so we had to fall back to the eager attention implementation in that case.

**Comment by xandery-geek** [comment]
Okay, I see. Thank you for your reply.
